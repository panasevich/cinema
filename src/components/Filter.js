import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';

export default class Filter extends PureComponent {
    handleFilter = (e) => {
        this.props.data(`FILTER_BY_${(e.target.innerText).replace( /\s/g, '').toUpperCase()}`)
    };
    render() {
        const {searchItems, title, alternate} = this.props;
        return (
            <div className={cn('search-by', {alternate})}>
                <div className="search-by-title">{title}</div>
                {searchItems.map((item, idx) =>
                    <div
                        className={cn('search-by-item', {active: item.active})}
                        key={`search-by-${idx}`}
                        onClick={this.handleFilter}
                    >
                        {item.title}
                    </div>
                )}
            </div>
        )
    }
}
Filter.propTypes = {
    title: PropTypes.string.isRequired,
    data: PropTypes.func.isRequired,
    searchItems: PropTypes.array.isRequired,
    alternate: PropTypes.bool,
};