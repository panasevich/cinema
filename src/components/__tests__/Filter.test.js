import React from 'react';
import Filter from '../Filter';
import { shallow } from '../../helpers/enzyme';
import renderer from 'react-test-renderer';

describe('Filter tests', () => {
    test('renders correctly', () => {
        const title = "Tile";
        const filters = [{title: 'filter1'},{title: 'filter2'},{title: 'filter3'},];
        const tree = renderer
            .create(<Filter title={title} searchItems={filters} onClick={()=>{}} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    test('renders filters', () => {
        const title = "Title";
        const filters = [{title: 'filter1'},{title: 'filter2'},{title: 'filter3'},];
        const wrapper = shallow(<Filter title={title} searchItems={filters} onClick={()=>{}} />);
        expect(wrapper.find('.search-by-item')).toBeDefined();
        expect(wrapper.find('.search-by-item')).toHaveLength(filters.length);
    });
    it('renders alternate filters design', () => {
        const title = "Title";
        const filters = [{title: 'filter1'},{title: 'filter2'},{title: 'filter3'},];
        const wrapper = shallow(<Filter title={title} searchItems={filters} alternate onClick={()=>{}} />);
        expect(wrapper.find('.alternate')).toBeDefined();
    });

});