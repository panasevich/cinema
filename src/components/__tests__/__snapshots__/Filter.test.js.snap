// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter tests renders correctly 1`] = `
<div
  className="search-by"
>
  <div
    className="search-by-title"
  >
    Tile
  </div>
  <div
    className="search-by-item"
    onClick={[Function]}
  >
    filter1
  </div>
  <div
    className="search-by-item"
    onClick={[Function]}
  >
    filter2
  </div>
  <div
    className="search-by-item"
    onClick={[Function]}
  >
    filter3
  </div>
</div>
`;
